tests/Basics/StringUtilsTest.cpp

```diff
@@ -348,3 +348,34 @@ TEST_F(StringUtilsTest, test_encodeURLComponent) {
   StringUtils::encodeURIComponent(result, ".!:$%@b013/-", 12);
   EXPECT_EQ("%20%25abcabc%20abc.!%3A%24%25%40b013%2F-", result);
 }
+
+TEST_F(StringUtilsTest, concatT) {
+  using namespace std::string_literals;
+  using namespace std::string_view_literals;
+  EXPECT_EQ("", StringUtils::concatT());
+  EXPECT_EQ("", StringUtils::concatT(""));
+  EXPECT_EQ("", StringUtils::concatT(""s));
+  EXPECT_EQ("", StringUtils::concatT(""sv));
+  EXPECT_EQ("42", StringUtils::concatT(42));
+  EXPECT_EQ("2.500000", StringUtils::concatT(2.5));
+  EXPECT_EQ("hello, world", StringUtils::concatT("hello", ", ", "world"));
+  EXPECT_EQ("cstr 42 stdstr view",
+            StringUtils::concatT("cstr ", 42, " stdstr "s, "view"sv));
+}
+
+TEST_F(StringUtilsTest, joinT) {
+  using namespace std::string_literals;
+  using namespace std::string_view_literals;
+  EXPECT_EQ("", StringUtils::joinT(", "sv));
+  EXPECT_EQ("", StringUtils::joinT(", "sv, ""));
+  EXPECT_EQ("", StringUtils::joinT(", "sv, ""s));
+  EXPECT_EQ("", StringUtils::joinT(", "sv, ""sv));
+  EXPECT_EQ("42", StringUtils::joinT(", "sv, 42));
+  EXPECT_EQ("2.500000", StringUtils::joinT(", "sv, 2.5));
+  EXPECT_EQ(", ", StringUtils::joinT(", "sv, "", ""));
+  EXPECT_EQ(", , ", StringUtils::joinT(", "sv, "", "", ""));
+  EXPECT_EQ("hello, world", StringUtils::joinT(", "sv, "hello", "world"));
+  EXPECT_EQ("hello, world", StringUtils::joinT(""sv, "hello", ", ", "world"));
+  EXPECT_EQ("cstr 42 stdstr view", StringUtils::joinT(""sv, "cstr ", 42, " stdstr "s, "view"sv));
+  EXPECT_EQ("cstr, 42, stdstr, view", StringUtils::joinT(", "sv, "cstr", 42, "stdstr"s, "view"sv));
+}
```

Comment:
maybe one could assert the capacity of the string to match the expected capacity.

Reply:
good idea.